library/src/main/scala/scala/scalajs/js/annotation/JSExportDescendentClasses.scala

```diff
@@ -17,4 +17,11 @@ package scala.scalajs.js.annotation
  *
  *  @see [[http://www.scala-js.org/doc/export-to-javascript.html Export Scala.js APIs to JavaScript]]
  */
-class JSExportDescendentClasses extends scala.annotation.StaticAnnotation
+class JSExportDescendentClasses extends scala.annotation.StaticAnnotation {
+  /** Constructor that allows ignoring descendants that cannot be exported
+   *
+   *  @param ignoreInvalidDescendants If true, descendants that cannot be exported
+   *      are silently ignored.
+   */
+  def this(ignoreInvalidDescendants: Boolean) = this()
```

Comment:
I know that trees are used to decode annotations rather than a semantic analysis, but this is still very weird, in particular for the ScalaDoc. 

I would make this the primary constructor.